### PR TITLE
Use best set/map diff in noms diff

### PR DESCRIPTION
**Pull request by kalman**
In 7c103344 I changed noms log to use left-right diff so that it
completes faster, but it changed noms diff to use left-right as well.
Noms diff is supposed to use best diff.

### CODE REVIEW

**Comment by kalman** [comment]
Hold up, I did something dumb.

**Comment by kalman** [comment]
Ok PTAL

File: cmd/noms/diff/diff.go [hunk 1]
```diff
@@ -17,16 +17,27 @@ type (
...
+}
+
 func shouldDescend(v1, v2 types.Value) bool {
 	kind := v1.Type().Kind()
 	return !types.IsPrimitiveKind(kind) && kind == v2.Type().Kind() && kind != types.RefKind
 }
 
-func Diff(w io.Writer, v1, v2 types.Value) error {
-	return diff(w, types.Path{}, nil, v1, v2)
+// Writes the Diff from |v1| to |v2| to |w|.
```
Comment (cmasone-attic) [inline cmd/noms/diff/diff.go]: If you start this comment with "Diff", godoc can pull the comment out for automatic documentation I believe.

`// Diff writes the diff from....`

File: cmd/noms/diff/diff.go [hunk 2]
```diff
@@ -17,16 +17,27 @@ type (
...
+// If |leftRight| is true then the left-diff diff is used for ordered sequences - see Diff vs DiffLeftRight in Set and Map.
```
Comment (cmasone-attic) [inline cmd/noms/diff/diff.go]: Do you mean '...then the left-right diff is used...'

**Comment by cmasone-attic** [comment]
LGTM w/comment nits

File: cmd/noms/diff/diff.go (hunk 1 above)
Comment (kalman) [inline cmd/noms/diff/diff.go]: done

File: cmd/noms/diff/diff.go (hunk 2 above)
Comment (kalman) [inline cmd/noms/diff/diff.go]: I do, thanks

**Pull request merged**